### PR TITLE
Add pagination support

### LINKED ISSUE #17: Posts after first page get lost
Instagram Basic API has [limitations](https://developers.facebook.com/docs/instagram-basic-display-api/#limitations) 

Ordering posts is not supported, therefore anything after page 1 is disregarded. For accounts with greater than 99 posts, I propose fetching all results and creating a collection from that using a limit.

Page one results are not the latest posts.

This Issue and PR #19 could allow users to sort posts by their timestamp.

### PR DESCRIPTION
**Pull request by spoyntersmith**
Please let me know what you think @michaeljoyner 
closes #17

### CODE REVIEW

File: src/Instagram.php [hunk 1]
```diff
@@ -92,14 +92,35 @@ public function fetchMedia(AccessToken $token, $limit = 20)
...
-        return collect($response['data'])
-            ->map(function($media) {
+        $page = true;
+        $collection = collect($response['data']);
+        $page_count = 1;
+        while ($page !== false) {
+            if (isset($response['paging'])) {
+                if (isset($response['paging']['next'])) {
+                    $page = $response['paging']['next'];
+                    var_dump(['page'.$page_count]);
```
Comment (michaeljoyner) [inline src/Instagram.php:102]: Can we remove the var_dump?

Comment (spoyntersmith) [reply src/Instagram.php]: 👍 Removed, thanks! I've also removed the `$page_count` variable

Comment (michaeljoyner) [reply src/Instagram.php]: Thanks, that's wonderful.

**Comment by michaeljoyner** [comment]
@spoyntersmith thanks so much for your work on this so far. I have one more thing which I think needs to be figured out. As far as I can tell, fetching the media will now continue to fetch as long as there is a next page available. However, if I only care about my most recent 10 or 20 posts, but I have many more, then as things stand now I will end up with many unnecessary requests. Or am I mistaken?

**Comment by spoyntersmith** [comment]
@michaeljoyner Agreed, I can modify this to compensate and use the `$limit` once again but still honour pagination.

For me as long as the timestamp is there then I can order posts by the Instagram timestamp rather than when I fetched.

**Comment by michaeljoyner** [comment]
Or we could make the limit parameter nullable, and if null limit is passed, then fetch all pages, possibly up to a point that is specified in the config.

File: src/Instagram.php [hunk 2]
```diff
@@ -76,9 +76,19 @@ public function refreshToken($token)
...
     {
-        $url = sprintf(self::MEDIA_URL_FORMAT, $token->user_id, self::MEDIA_FIELDS, $limit, $token->access_code);
+        $queryLimit = $limit; //optimal number of requests
+
+        if ($limit === null || $limit > 1000) {
+            $queryLimit = 100;
+        } elseif ($limit <= 100) {
+            $queryLimit = $limit;
+        } elseif ($limit > 100) {
+            $queryLimit = (int) ceil( $queryLimit/ ceil($queryLimit/100) );
```
Comment (spoyntersmith) [inline src/Instagram.php:88]: calculates optimal $queryLimit unless over 1000 in which case will be 100 per request

File: src/Instagram.php [hunk 3]
```diff
@@ -76,9 +76,19 @@ public function refreshToken($token)
         return $this->http->get($url);
     }
 
-    public function fetchMedia(AccessToken $token, $limit = 20)
+    public function fetchMedia(AccessToken $token, $limit = null)
```
Comment (spoyntersmith) [inline src/Instagram.php:79]: still honours the default 20 from `$profile->feed()` or `$profile->refreshFeed()` in Instagram.php 

unless `null` is passed through

**Comment by spoyntersmith** [comment]
@michaeljoyner Latest commit will for example only make 3 requests if limit passed is 242

**Comment by michaeljoyner** [comment]
@spoyntersmith your solution was great thanks, I have gone with it. I added a commit to break up the fetchMedia method a bit, and to add a test, sorry if I made things a bit messy. If you are happy with the way things are I will merge this PR

**Comment by spoyntersmith** [comment]
@michaeljoyner I think this is much cleaner than my commit, the requests are now wrapped in the try-catch too.

Thanks again!  😊👍

The two areas of the docs that probably need updating are **Getting the feed** and **Updating the feed**, happy to help!

**Comment by spoyntersmith** [comment]
One last thing, I think there's an issue splicing with null as the limit...

**Comment by michaeljoyner** [comment]
Thanks for reminding me about the docs, I probably works have forgotten. Nice catch with the take method, I think that is much better. However, seeing as $limit could be null, it is probably better to use take($limit ?? $collection->count()) to be a bit more clear

**Comment by spoyntersmith** [comment]
@michaeljoyner That's done and i've added support for `ignore_videos` otherwise the last collection map removes them and you get a smaller collection. `->values()` here ensures that the keys will be normal (0, 1, 2, 3, ..., 19)

At this point I would review the `getPageSize()`, there's two options here. 
When ignore videos is set you could set the query limit to 100 to ensure that enough images are returned in the same request or, leave as is and it will make the requests until it has enough `$collection` items.

Would a default query size of 100 be unreasonable?

**Comment by michaeljoyner** [comment]
@spoyntersmith That is great, thank you. Your work on this is really appreciated. For the page size issue, I think let's just leave it as is, I have included notes about it in the readme. People can choose to fetch more if they feel it is necessary.

I think this PR is ready to be merged, but I will wait until you have had a chance to check if you're fine with everything.

**Comment by spoyntersmith** [comment]
I've ran checks and everything is passing for me.
Ready to merge @michaeljoyner 🚀

**Comment by michaeljoyner** [comment]
@spoyntersmith Thanks again for your work on this, it is most appreciated.

**Pull request merged**